### ISSUE TITLE
`unit` not showing-up in .ttl

### ISSUE THREAD

**Issue by sebastiankb** [w3c/wot-thing-description#1450]
The context file maps the term `unit` to `https://schema.org/unitCode`. However, the term `unit` itself does not show up in the TD or jsonschema ttl file, which leads to a disconnection and gives the impression that no `unit` definition is possible for Thing Descriptions. 

Proposal: involve unit definition in the TD ttl that points to `https://schema.org/unitCode`

**Comment by sebastiankb** [comment]
@ethieblin fyi

**Comment by ethieblin** [comment]
We can propose a PR to add a new ObjectProperty `td:unit` which has a owl:sameAs to the schema:unitCode.
Would that be OK ?

**Comment by fennibay** [comment]
I question in general if we need to define td:unit. Basically, [TD context](https://www.w3.org/2022/wot/td/v1.1) maps unit to schema:unitCode, and schema.org provides the definition. This is the same for instance with DataSchema, which is mapped to jsonschema:DataSchema by the context, and defined in jsonschema ontology.

OTOH, we don't have an RDF statement that says that jsonschema:DataSchema can have a schema:unitCode, so maybe we need this definition somewhere: // I propose jsonschema ontology

```ttl
schema:unitCode schema:domainIncludes jsonschema:DataSchema .
```

Another observation is that the current TD context for unit doesn't match the [definition on schema.org](https://schema.org/unitCode), which says: 

>The unit of measurement given using the UN/CEFACT Common Code (3 characters) or a URL. Other codes than the UN/CEFACT Common Code may be used with a prefix followed by a colon.

So I think the context definition should be changed from this:

```json
        "unit": {
          "@id": "schema:unitCode",
          "@type": "@vocab"
        }
```

to this:

```json
        "unit": {
          "@id": "schema:unitCode",
          "@type": "@vocab",
          "@context": {
            "@vocab": "ontology_link_toUNCEFACT_common_code"
          }
        }
```

This way if somebody uses the UN/CEFACT common code, it will also resolve to a proper IRI.

p.s.: If we have to define td:unit, then I'd recommend owl:equivalentProperty or rdfs:subPropertyOf instead of owl:sameAs. Reasoning: https://www.w3.org/TR/skos-reference/#L4858

@mhusm

**Comment by mhusm** [comment]
> OTOH, we don't have an RDF statement that says that jsonschema:DataSchema can have a schema:unitCode, so maybe we need this definition somewhere: // I propose jsonschema ontology
> 
> ```turtle
> schema:unitCode schema:domainIncludes jsonschema:DataSchema .
> ```

Agreed, this was my main reason for bringing it up internally at Siemens. I also like the proposed solution.

> 
> Another observation is that the current TD context for unit doesn't match the [definition on schema.org](https://schema.org/unitCode), which says:
> 
> > The unit of measurement given using the UN/CEFACT Common Code (3 characters) or a URL. Other codes than the UN/CEFACT Common Code may be used with a prefix followed by a colon.
> 
> So I think the context definition should be changed from this:
> 
> ```json
>         "unit": {
>           "@id": "schema:unitCode",
>           "@type": "@vocab"
>         }
> ```
> 
> to this:
> 
> ```json
>         "unit": {
>           "@id": "schema:unitCode",
>           "@type": "@vocab",
>           "@context": {
>             "@vocab": "ontology_link_toUNCEFACT_common_code"
>           }
>         }
> ```
> 
> This way if somebody uses the UN/CEFACT common code, it will also resolve to a proper IRI.

The current TD context processes something like `"unit": "CEL"` into `_:b1 <http://schema.org/unitCode> <https://www.w3.org/2019/wot/json-schema#CEL> .`  @fennibay's proposal would fix the URL, however, I couldn't find a complete  UN/CEFACT ontology that could be referenced. For our use cases, we will most likely use QUDT IRIs, which is possible with both the current TD context and @fennibay's proposal. 

> p.s.: If we have to define td:unit, then I'd recommend owl:equivalentProperty or rdfs:subPropertyOf instead of owl:sameAs. Reasoning: https://www.w3.org/TR/skos-reference/#L4858

Yes, this is the way to go. owl:sameAs is defined for individuals, not properties or classes.

**Comment by Murloc6** [comment]
We agree with what is proposed above, but in this case all the units should come from this ontology (if we can find one). As discussed in #1202 sometimes it is useful to have the unit as human-readable Literal. This is why we proposed to create separate entries for these two situations (see #1210).

We could also imagine a unit definition, which could take the following form:
```
"@context": {
    "schema": "http://schema.org/",
    "td": "https://www.w3.org/2019/wot/td#",
    "unit": {
      "@id": "schema:unitCode",
      "@type": "@vocab",
      "@context": {
        "@vocab": "http://my-unit-ontology#",
        "names": {
          "@id": "td:unitName",
          "@container": "@language"
        },
        "symbol": "td:unitSymbol"
      }
    }
  }
```
This context can handle the three following cases:
1. `"unit": "CMT"` (UN/CEFACT form) [playground](https://json-ld.org/playground/#startTab=tab-nquads&json-ld=%7B%22%40context%22%3A%7B%22schema%22%3A%22http%3A%2F%2Fschema.org%2F%22%2C%22td%22%3A%22https%3A%2F%2Fwww.w3.org%2F2019%2Fwot%2Ftd%23%22%2C%22unit%22%3A%7B%22%40id%22%3A%22schema%3AunitCode%22%2C%22%40type%22%3A%22%40vocab%22%2C%22%40context%22%3A%7B%22%40vocab%22%3A%22http%3A%2F%2Fmy-unit-ontology%23%22%2C%22names%22%3A%7B%22%40id%22%3A%22td%3AunitName%22%2C%22%40container%22%3A%22%40language%22%7D%2C%22symbol%22%3A%22td%3AunitSymbol%22%7D%7D%7D%2C%22%40id%22%3A%22urn%3Asomething%22%2C%22unit%22%3A%22CMT%22%7D)
2. `"unit": "qudt:cm"` (URI using whatever ontology, this example uses qudt) [playground](https://json-ld.org/playground/#startTab=tab-nquads&json-ld=%7B%22%40context%22%3A%7B%22schema%22%3A%22http%3A%2F%2Fschema.org%2F%22%2C%22td%22%3A%22https%3A%2F%2Fwww.w3.org%2F2019%2Fwot%2Ftd%23%22%2C%22unit%22%3A%7B%22%40id%22%3A%22schema%3AunitCode%22%2C%22%40type%22%3A%22%40vocab%22%2C%22%40context%22%3A%7B%22%40vocab%22%3A%22http%3A%2F%2Fmy-unit-ontology%23%22%2C%22names%22%3A%7B%22%40id%22%3A%22td%3AunitName%22%2C%22%40container%22%3A%22%40language%22%7D%2C%22symbol%22%3A%22td%3AunitSymbol%22%7D%7D%7D%2C%22%40id%22%3A%22urn%3Asomething%22%2C%22unit%22%3A%22http%3A%2F%2Fmy-other-unit-ontology%23Centimeters%22%7D)
3. an anonymous unit on which we define its names and symbols: [playground](https://json-ld.org/playground/#startTab=tab-nquads&json-ld=%7B%22%40context%22%3A%7B%22schema%22%3A%22http%3A%2F%2Fschema.org%2F%22%2C%22td%22%3A%22https%3A%2F%2Fwww.w3.org%2F2019%2Fwot%2Ftd%23%22%2C%22unit%22%3A%7B%22%40id%22%3A%22schema%3AunitCode%22%2C%22%40type%22%3A%22%40vocab%22%2C%22%40context%22%3A%7B%22%40vocab%22%3A%22http%3A%2F%2Fmy-unit-ontology%23%22%2C%22names%22%3A%7B%22%40id%22%3A%22td%3AunitName%22%2C%22%40container%22%3A%22%40language%22%7D%2C%22symbol%22%3A%22td%3AunitSymbol%22%7D%7D%7D%2C%22%40id%22%3A%22urn%3Asomething%22%2C%22unit%22%3A%7B%22names%22%3A%7B%22fr%22%3A%22centimetres%22%2C%22en%22%3A%22centimeters%22%7D%2C%22symbol%22%3A%22cm%22%7D%7D)
```
"unit": {
    "names": {
      "fr": "centimetres",
      "en": "centimeters"
    },
    "symbol": "cm"
  }
```

In this example, we use the `schema:unitCode` property, but it cannot cover case number 3.
Then we could create a new property like `td:unit` and a class `td:Unit` for example.

**Comment by sebastiankb** [comment]
>This is why we proposed to create separate entries for these two situations (see https://github.com/w3c/wot-thing-description/issues/1210).

Yes, this seems to be useful. As mentioned in this issue this will be the plan for TD 2.0.

>We could also imagine a unit definition, which could take the following form:

> We could also imagine a unit definition, which could take the following form:
> 
> ```
> "@context": {
>     "schema": "http://schema.org/",
>     "td": "https://www.w3.org/2019/wot/td#",
>     "unit": {
>       "@id": "schema:unitCode",
>       "@type": "@vocab",
>       "@context": {
>         "@vocab": "http://my-unit-ontology#",
>         "names": {
>           "@id": "td:unitName",
>           "@container": "@language"
>         },
>         "symbol": "td:unitSymbol"
>       }
>     }
>   }
> ```
 
Looks like a good solution to me. @mhusm @fennibay what do you think?

**Comment by sebastiankb** [comment]
I just realize that the 3. option

>3. an anonymous unit on which we define its names and symbols 

should be not implemented since this causes a validation conflict since the `unit` is expected to be a `string` only.
